### PR TITLE
Add configuration for CLA Assistant GitHub Action

### PR DESCRIPTION
**Pull request by michaelosthege**
Internal ticket: https://jugit.fz-juelich.de/IBG-1/rse/proxy-issues/-/issues/17

### CODE REVIEW

File: .github/workflows/cla.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+              Like many open-source projects we ask that you sign our [Contributor License Agreement]($PATH_TO_CLA) before we can accept your contribution.
+              To sign, please post a comment based on the following template 👇
+          custom-pr-sign-comment: >
+              I have read the CLA Document and I hereby sign the CLA.
+
+              + [ ] The JuBiotech maintainers know my real name.
+
+              At least one of the following two applies:
+
+              + [ ] The JuBiotech maintainers know my current employer.
```
Comment (schmoelder) [inline .github/workflows/cla.yml:51]: What about people who do this in their free time or students?

Comment (schmoelder) [reply .github/workflows/cla.yml:51]: Nevermind, I misread the line below.

File: .github/workflows/cla.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: "CLA Assistant"
+on:
+  issue_comment:
+    types: [created]
```
Comment (schmoelder) [inline .github/workflows/cla.yml:4]: Is it really required for issues or only for PRs?

Comment (michaelosthege) [reply .github/workflows/cla.yml]: I agree and took it out 👍

**Comment by ronald-jaepel** [comment]
I've added the yml to a[ dumy repo to test](https://github.com/modsim/CADET-CLA-test-2/pull/5) and found a couple of things. 

Namely, the $PATH_TO_CLA syntax doesn't work, we need to use ${{ env.PATH_TO_CLA }}
Also, I can't get the token-access to work. Will report back when that works

**Comment by michaelosthege** [comment]
> So, the bot does a full isequal check before accepting sign comments (meaning the comment has to be exactly equal to the value set in custom-pr-sign-comment). We therefore can not add additional info into the single sign comment. Other then that, everything works now in my mock test.

thanks for testing! Did you check with the YML version from this PR?
Because (a few hours ago) I modified that equals check to a "contains".

**Comment by ronald-jaepel** [comment]
I tested with the yml from here. I think the code you modified only decides if the comment triggers a CLA-Assistant action, not what the action is going to be, AFAIK [this](https://github.com/contributor-assistant/github-action/blob/a895a435fcce79ecf28fbce61a4ef0f0dabc9853/src/pullrequest/signatureComment.ts#L62) implements the comparison to the reference comment.

Anyways, here are my suggested changes. 


```
name: "CLA Assistant"
on:
  pull_request_target:
    types: [opened,closed,synchronize]

# explicitly configure permissions, in case your GITHUB_TOKEN workflow permissions are set to read-only in repository settings
permissions:
  actions: write
  contents: write
  pull-requests: write
  statuses: write

jobs:
  CLAAssistant:
    runs-on: ubuntu-latest
    steps:
      - name: "CLA Assistant"
        if: (github.event.comment.body == 'recheck' || contains(github.event.comment.body, 'I have read the CLA Document and I hereby sign the CLA')) || github.event_name == 'pull_request_target'
        uses: contributor-assistant/github-action@v2.3.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          # The below token has repo scope for the project configured further below to store signatures.
          PERSONAL_ACCESS_TOKEN: ${{ secrets.JUBIOTECH_CLA_SIGNATURES }}
          PATH_TO_CLA: 'https://github.com/jubiotech/detl/blob/main/CLA.md'
        with:
          path-to-signatures: 'signatures_detl.json'
          path-to-document: $PATH_TO_CLA
          # branch should not be protected
          branch: 'main'
          allowlist: dependabot[bot]

          # the followings are the optional inputs - If the optional inputs are not given, then default values will be taken
          # 👇 the remote organization and repo where the signatures should be stored
          remote-organization-name: jubiotech
          remote-repository-name: cla-signatures
          #create-file-commit-message: 'For example: Creating file for storing CLA Signatures'
          #signed-commit-message: 'For example: $contributorName has signed the CLA in $owner/$repo#$pullRequestNo'
          custom-notsigned-prcomment: >
              Thank you for your submission, we really appreciate it!
            
              Like many open-source projects we ask that you sign our [Contributor License Agreement](${{ env.PATH_TO_CLA }}) before we can accept your contribution.
              To sign, please post two separate comments based on the following template 👇


              1. Comment:
              
              ___
              
              \- [ ] The JuBiotech maintainers know my real name.

              At least one of the following two applies:

              \- [ ] The JuBiotech maintainers know my current employer.
              
              \- [ ] I am making not making this contribution of behalf of my current employer.

              ___

              2. Comment:
              
          custom-pr-sign-comment: >
              I have read the CLA Document and I hereby sign the CLA.

          #custom-allsigned-prcomment: 'pull request comment when all contributors has signed, defaults to **CLA Assistant Lite bot** All Contributors have signed the CLA.'
          # 👇 prevent contributors from revoking signatures after the merge (this is also the default)
          lock-pullrequest-aftermerge: true
          #use-dco-flag: true - If you are using DCO instead of CLA

```

**Comment by michaelosthege** [comment]
Thanks @ronald-jaepel I applied your changes and reversed my change in the `if` condition.

**Review by ronald-jaepel — APPROVED** [review]
looks good to me

File: .github/workflows/cla.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+              ----
+
+              \- [ ] The JuBiotech maintainers know my real name.
+
+              At least one of the following two applies:
+
+              \- [ ] The JuBiotech maintainers know my current employer.
+
+              \- [ ] I am making not making this contribution of behalf of my current employer.
```
Comment (mbeyss) [inline .github/workflows/cla.yml:54]:
```suggestion
              \- [ ] I am making not making this contribution on behalf of my current employer.

```

**Review by mbeyss — CHANGES_REQUESTED** [review]

File: .github/workflows/cla.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+              ----
+
+              \- [ ] The JuBiotech maintainers know my real name.
+
+              At least one of the following two applies:
+
+              \- [ ] The JuBiotech maintainers know my current employer.
+
+              \- [ ] I am making not making this contribution on behalf of my current employer.
```
Comment (schmoelder) [inline .github/workflows/cla.yml:54]: Shouldnt this be 

```
I am not making this contribution on behalf of my current employer.
```
?

Comment (michaelosthege) [reply .github/workflows/cla.yml:54]: No, if the contribution is made in the contributors free time, their current employer is irrelevant. They might also be self- or unemployed. Therefore only one of the two task items must apply.

Comment (schmoelder) [reply .github/workflows/cla.yml:54]: I get that, I just think there is one *making* too many in the second point.

Comment (michaelosthege) [reply .github/workflows/cla.yml:54]:
```suggestion
              \- [ ] I am making not this contribution on behalf of my current employer.

```
ah of course, sorry my in-mind-diffing was incomplete

Comment (lieres) [reply .github/workflows/cla.yml]: I really think we need to consult Alexander Storm here. In my view, it makes a difference if somehone has an employer of if he/she made the contribution during working hours. To make things more complex, contributions in "free time" might still require permission of thge employer (or not). This is not our business but it must be clear to the contributor which option to choose. Some contributors (like Sam) have multiple accounts to clarify on/off work contributions.

Comment (michaelosthege) [reply .github/workflows/cla.yml]: Okay, then I will summarize the proposed procedure, texts & mockup screenshots, and consult him by email (cc y'all).

**Pull request merged**